portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java

```diff
@@ -0,0 +1,66 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.UserStatus;
+import com.google.sps.data.Params;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that sends authentication status and associated login/logout links to the client.*/
+@WebServlet("/authenticate-user")
+public class AuthenticationServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType(Params.JSON_CONTENT_TYPE);
+    UserService userService = UserServiceFactory.getUserService();
+    UserStatus status;
+
+    // Create UserStatus based on whether the user is logged in or not.
+    if (userService.isUserLoggedIn()) {
+      // The user IS logged in.
+      String logoutUrl = userService.createLogoutURL(Params.BASE_URL_PATH);
+      status = new UserStatus(/** isLoggedIn= */ true, logoutUrl);
```

Comment:
If isLoggedIn is only ever going to be true or false, consider changing the input to an enum that is defined inside of UserStatus.

```
class UserStatus {
  static enum AuthState {
    LOGGED_IN = 0,
    LOGGED_OUT = 1,
  }
}
```